src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double SIZE_MULTIPLICATION = 1.5;
+    private int arrayLength = DEFAULT_CAPACITY;
+    private T[] array = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size;
+
+    public ArrayList() {
+    }
 
     @Override
     public void add(T value) {
-
+        try {
+            if (size == array.length) {
+                array = reSize(array);
+            }
+            array[size] = value;
+            size++;
+        } catch (NullPointerException e) {
```

Comment:
catching RuntimeExceptions is bad practice - please remove this from the code

Reply:
done